File: src/net/theharrisoncrafter/God.java
package net.theharrisoncrafter;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.logging.Logger;

import net.theharrisoncrafter.Commands.Nickname;

import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;
import org.bukkit.plugin.PluginDescriptionFile;
import org.bukkit.plugin.java.JavaPlugin;

public class God extends JavaPlugin{
	Logger log;
	public static PluginDescriptionFile pdf;
	public static FileConfiguration config;
	
	@Override
	public void onLoad() {
		log = this.getLogger();
		pdf = this.getDescription();
		
		this.saveDefaultConfig();
		config = this.getConfig();
	}
	
	@Override
	public void onEnable(){
	}
	
	@Override
	public void onDisable(){
	}
	
	public boolean onCommand(final CommandSender sender, final Command command, final String label, final String[] args){
		try{
			String c = command.getName();
			return ((CommandExecutor) Class.forName("net.theharrisoncrafter.Commands." + c.toUpperCase().charAt(0) + c.toLowerCase().substring(1)).getConstructor().newInstance()).onCommand(sender, command, label, args);
		}catch(final Exception e){
			sender.sendMessage("Something went wrong (" + e.getClass() + "): " + e.getMessage() + "!(" + e.getCause() + ")");
			sender.sendMessage("Please report this problem to theharrisoncrafter@gmail.com with this error and a list of the steps you took to get this error.");
			e.printStackTrace();
		}
		return false;
	}
	
	public String getVertsion(){
		return pdf.getVersion();
	}
}

File: src/main/kotlin/co/orre/godcraft/God.kt
package co.orre.godcraft

import co.orre.godcraft.Commands.*
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.plugin.java.JavaPlugin

class God : JavaPlugin() {

    val configuration = Configuration(this)

    override fun onEnable() {
        this.saveDefaultConfig()
        config.options().copyDefaults(true)
        config.set("version", description.version)
        saveConfig()
        configuration.load()

        getCommand("boom").executor = Boom(this)
        getCommand("burn").executor = Burn(this)
        getCommand("feed").executor = Feed(this)
        getCommand("getcrafting").executor = GetCrafting(this)
        getCommand("getenchanting").executor = GetEnchanting(this)
        getCommand("getenderchest").executor = GetEnderChest(this)
        getCommand("getinventory").executor = GetInventory(this)
        getCommand("heal").executor = Heal(this)
        getCommand("lightning").executor = Lightning(this)
        getCommand("murder").executor = Murder(this)
        getCommand("nickname").executor = Nickname(this)
        getCommand("ping").executor = Ping(this)
        getCommand("setfoodlevel").executor = SetFoodLevel(this)
        getCommand("sethealth").executor = SetHealth(this)
        getCommand("setlevel").executor = SetLevel(this)
        getCommand("setmaxair").executor = SetMaxAir(this)
        getCommand("setmaxhealth").executor = SetMaxHealth(this)
        getCommand("starve").executor = Starve(this)
        getCommand("throw").executor = Throw(this)
        getCommand("troll").executor = Troll(this)
    }

    override fun onDisable() {}

    override fun onCommand(sender: CommandSender?, command: Command?, label: String?, args: Array<String>?): Boolean {
        try {
            val c = command!!.name
            return (Class.forName("net.theharrisoncrafter.Commands." + c.toUpperCase()[0] + c.toLowerCase().substring(1)).getConstructor().newInstance() as CommandExecutor).onCommand(sender, command, label, args)
        } catch (e: Exception) {
            sender!!.sendMessage("Something went wrong (" + e.javaClass + "): " + e.message + "!(" + e.cause + ")")
            sender.sendMessage("Please report this problem to theharrisoncrafter@gmail.com with this error and a list of the steps you took to get this error.")
            e.printStackTrace()
        }

        return false
    }

    fun logDebug(msg: String) {
        if (!configuration.DEBUG) return
        logger.info(msg)
    }
}